### PR TITLE
feat: add namespace read API

### PR DESCRIPTION
**Pull request by hperl**
This PR adds an API (gRPC & HTTP) to retrieve the list of namespaces.

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md).
- [x] I have referenced an issue containing the design document if my change
      introduces a new feature.
- [x] I am following the
      [contributing code guidelines](../blob/master/CONTRIBUTING.md#contributing-code).
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security vulnerability, I
      confirm that I got the approval (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [x] I have added tests that prove my fix is effective or that my feature
      works.
- [ ] I have added or changed [the documentation](https://github.com/ory/docs).

### CODE REVIEW

File: internal/e2e/grpc_client_test.go [hunk 1]
```diff
@@ -28,6 +28,15 @@ type grpcClient struct {
 	ctx                                      context.Context
 }
 
+func (g *grpcClient) queryNamespaces(t require.TestingT) (apiResponse ketoapi.GetNamespacesResponse) {
+	client := rts.NewNamespacesServiceClient(g.readConn(t))
+	res, err := client.ListNamespaces(g.ctx, &rts.ListNamespacesRequest{})
+	require.NoError(t, err)
+	require.NoError(t, convert(res, &apiResponse))
```
[the remark this answers is not in the document]
Comment (hperl) [reply internal/e2e/grpc_client_test.go:35]: The structs are not exactly the same, they just have the same `json` tags. For example, the protobuf struct has extra private fields:
```go
type ListNamespacesResponse struct {
	state         protoimpl.MessageState
	sizeCache     protoimpl.SizeCache
	unknownFields protoimpl.UnknownFields

	Namespaces []*Namespace `protobuf:"bytes,1,rep,name=namespaces,proto3" json:"namespaces,omitempty"`
}
```

File: internal/e2e/testcases_test.go [hunk 2]
```diff
@@ -20,6 +20,15 @@ func runCases(c client, m *namespaceTestManager) func(*testing.T) {
 	return func(t *testing.T) {
 		c.waitUntilLive(t)
 
+		t.Run("case=list namespaces", func(t *testing.T) {
+			first := namespace.Namespace{Name: "my namespace"}
+			second := namespace.Namespace{Name: "my other namespace"}
+			m.add(t, &first, &second)
+
+			resp := c.queryNamespaces(t)
+			assert.GreaterOrEqual(t, len(resp.Namespaces), 2)
```
[the remark this answers is not in the document]
Comment (hperl) [reply internal/e2e/testcases_test.go:29]: done :)

File: internal/e2e/sdk_client_test.go [hunk 3]
```diff
@@ -20,6 +20,7 @@ type sdkClient struct {
 	wc httpclient.WriteApi
 	mc httpclient.MetadataApi
 	sc httpclient.SyntaxApi
+	nc httpclient.NamespacesApi
```
[the remark this answers is not in the document]
Comment (hperl) [reply internal/e2e/sdk_client_test.go]: I refactored it a bit. PTAL :)

**Review by zepatrik — APPROVED** [review]
Nice :+1:

**Pull request merged**